beacon-chain/core/state/transition_test.go

```diff
@@ -394,251 +393,84 @@ func TestProcessBlock_PassesProcessingConditions(t *testing.T) {
-		LatestCrosslinks:   crosslinkRecord,
-		LatestRandaoMixes:  randaoHashes,
-		LatestActiveIndexRoots: make([][]byte,
-			params.BeaconConfig().LatestActiveIndexRootsLength),
-		LatestSlashedBalances: make([]uint64,
-			params.BeaconConfig().LatestSlashedExitLength),
-	}
-
-	_, err := state.ProcessEpoch(context.Background(), newState)
-	if err != nil {
-		t.Errorf("Expected epoch transition to pass processing conditions: %v", err)
+func TestProcessEpoch_CantGetTgtAttsPrevEpoch(t *testing.T) {
+	atts := []*pb.PendingAttestation{{Data: &pb.AttestationData{TargetEpoch: 1}}}
+	_, err := state.ProcessEpoch(context.Background(), &pb.BeaconState{CurrentEpochAttestations: atts})
+	if !strings.Contains(err.Error(), "could not get target atts prev epoch") {
+		t.Fatal("Did not receive wanted error")
 	}
 }
 
-func TestProcessEpoch_PreventsRegistryUpdateOnNilBlock(t *testing.T) {
-	t.Skip()
-	// TODO(#2307) unskip after ProcessCrosslinks is finished
-	featureconfig.InitFeatureConfig(&featureconfig.FeatureFlagConfig{
-		EnableCrosslinks: false,
-	})
-	var validatorRegistry []*pb.Validator
-	for i := uint64(0); i < 10; i++ {
-		validatorRegistry = append(validatorRegistry,
-			&pb.Validator{
-				ExitEpoch: params.BeaconConfig().FarFutureEpoch,
-			})
-	}
-	validatorBalances := make([]uint64, len(validatorRegistry))
-	for i := 0; i < len(validatorBalances); i++ {
-		validatorBalances[i] = params.BeaconConfig().MaxDepositAmount
-	}
-
-	var attestations []*pb.PendingAttestation
-	for i := uint64(0); i < params.BeaconConfig().SlotsPerEpoch*2; i++ {
-		attestations = append(attestations, &pb.PendingAttestation{
-			Data: &pb.AttestationData{
-				Slot:                     i + params.BeaconConfig().SlotsPerEpoch,
-				Shard:                    2,
-				JustifiedEpoch:           1,
-				JustifiedBlockRootHash32: []byte{0},
-			},
-			InclusionDelay: i + params.BeaconConfig().SlotsPerEpoch + 1,
-		})
-	}
-
-	var blockRoots [][]byte
-	for i := uint64(0); i < params.BeaconConfig().LatestBlockRootsLength; i++ {
-		blockRoots = append(blockRoots, []byte{byte(i)})
-	}
-
-	var randaoHashes [][]byte
-	for i := uint64(0); i < params.BeaconConfig().SlotsPerEpoch; i++ {
-		randaoHashes = append(randaoHashes, []byte{byte(i)})
-	}
-
-	crosslinkRecord := make([]*pb.Crosslink, 64)
-	newState := &pb.BeaconState{
-		Slot:               params.BeaconConfig().SlotsPerEpoch + 1,
-		LatestAttestations: attestations,
-		Balances:           validatorBalances,
-		ValidatorRegistry:  validatorRegistry,
-		LatestBlockRoots:   blockRoots,
-		LatestCrosslinks:   crosslinkRecord,
-		LatestRandaoMixes:  randaoHashes,
-		LatestActiveIndexRoots: make([][]byte,
-			params.BeaconConfig().LatestActiveIndexRootsLength),
-		LatestSlashedBalances: make([]uint64,
-			params.BeaconConfig().LatestSlashedExitLength),
-		ValidatorRegistryUpdateEpoch: 0,
-		FinalizedEpoch:               1,
-	}
+func TestProcessEpoch_CantGetTgtAttsCurrEpoch(t *testing.T) {
+	e := params.BeaconConfig().SlotsPerEpoch
```

Comment:
Why is this the case?

e = slots per epoch? So this is the 8th epoch? why?

Reply:
it's the 1st epoch right?

```go
&pb.BeaconState{Slot:e} // Slot: 64
```

Reply:
can you name it `s` or `slot`? `e` has been shorthand for `epoch` elsewhere in the code